bip-reservedbits.mediawiki

```diff
@@ -0,0 +1,51 @@
+<pre>
+  BIP: ?
+  Title: Reserved nVersion bits for general purpose use
+  Author: BtcDrak <btcdrak@gmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-????
+  Status: Draft
+  Type: Informational
+  Created: 2018-03-01
+  License: BSD-3-Clause
+           CC0-1.0
+</pre>
+
+==Abstract==
+
+This BIP reserves 16 bits of the block header nVersion field for general purpose use and removes their meaning for the purpose of version bits soft-fork signalling.
+
+==Motivation==
+
+There are a variety of things that miners may desire to use some of the nVersion field bits for. However, due to their use to coordinate miner activated soft-forks, full node software will generate false warnings about unknown soft forks if those bits are used for non soft fork signalling purposes. By reserving bits from the nVersion field for general use, node software can be updated to ignore those bits and therefore will not emit false warnings. Reserving 16 bits for general use leaves enough for 13 parallel soft-forks using version bits.
+
+==Example Uses==
+
+The following are example cases that would benefit from using some of the bits from the nVersion field. This list is not exhaustive.
+
+Bitcoin mining hardware currently can exhaust the 32 bit nonce field in less than 200ms requiring the controller to distribute new jobs very frequently to each mining chip consuming a lot of bandwidth and CPU time. This can be greatly reduced by rolling more bits. Rolling too many bits from nTime is not ideal because it may distort the timestamps over a longer period.
+
+Version-rolling AsicBoost requires two bits from the nVersion field to calculate 4-way collisions. Any two bits can be used and mining equipment can negotiate which bits are to be used with mining pools via the Stratum "version-rolling" extension.
+
+==Specification==
+
+Sixteen bits from the block header nVersion field, starting from 13 and ending at 28 inclusive (0x1fffe000), are reserved for general use and removed from BIP8 and BIP9 specifications. A mask of 0xe0001fff should be applied to nVersion bits so bits 13-28 inclusive will be ignored for soft-fork signalling and unknown soft-fork warnings.
+
+This specification does not reserve specific bits for specific purposes.
+
+==Backwards Compatibility==
+
+This proposal is backwards compatible, and does not require a soft fork to implement.
+
+==References==
+
+  - [[bip-0008.mediawiki|BIP8]]
```

Comment:
Please format using a `*` at the beginning of the line so this list renders correctly in mediawiki markup. Eg. code should be:
```
* [[bip-0008.mediawiki|BIP8]]
* [[bip-0009.mediawiki|BIP9]]
* [https://arxiv.org/pdf/1604.00575.pdf AsicBoost white paper]
* [https://github.com/BlockheaderNonce2/bitcoin/wiki nNonce2 proposal]
* [http://stratumprotocol.org/ Stratum protocol extension for version-rolling]
```